File: designPattern/Bridge/BridgeJava/src/StringDisplayImpl.java
public class StringDisplayImpl extends DisplayImpl{
    String names;
    int limit;
    public StringDisplayImpl(String names){
        this.names = names;
        limit = names.getBytes().length;
    }
    @Override
    public void rawOpen() {
        printLine();
    }

    @Override
    public void rawPrint() {
        System.out.println("|" + names + "|");
    }

    @Override
    public void rawClose() {
        printLine();
    }

    private void printLine(){
        System.out.print("+" );
        for(int i =0; i < limit; i++)
            System.out.print("-");
        System.out.println("+");
    }
}

File: designPattern/Bridge/BridgePython/StringDisplayimpl.py
from Displayimpl import Displayimpl


class StringDisplayimpl(Displayimpl):

    def __init__(self, str):
        self.str = str
        self.limit = len(str)

    def rawOpen(self):
        self.printLine()

    def rawPrint(self):
        print("|"+self.str+"|")

    def rawClose(self):
        self.printLine()

    def printLine(self):
        print("+", end="")
        for i in range(self.limit):
            print("-", end="")
        print("+")
